model/project_ref.go

```diff
@@ -2203,6 +2244,38 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
+// ValidateProjectCreation returns an error if the total or owner/repo project limit set by the admin settings
+// has been reached. If the project is not enabled, it will grip a warning but not error.
+func ValidateProjectCreation(config *evergreen.Settings, projectRef *ProjectRef) error {
+	catcher := grip.NewBasicCatcher()
+	allEnabledProjects, err := GetNumberOfProjects()
+	if err != nil {
+		return errors.Wrap(err, "getting number of projects")
+	}
+	if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
+	}
+	enabledOwnerRepoProjects, err := GetNumberOfProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		return errors.Wrapf(err, "getting number of projects for '%s'/'%s'", projectRef.Owner, projectRef.Repo)
+	}
+	if enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit && !config.ProjectCreation.IsException(projectRef.Owner, projectRef.Repo) {
```

Comment:
Can we check exception before counting num enabled projects so we don't do unnecessary work?